### PR TITLE
Ack fixes

### PR DESCRIPTION
**Pull request by stoader**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?

- Access ESS service at the endpoints listed in here: https://www.alibabacloud.com/help/doc-detail/25927.htm
- Fix the logic that determines what node pool a node belongs to
- After tiller is upgraded wait until the old tiller pod stops and the new one comes up before considering it as complete

### Checklist

- [x] Implementation tested (with at least one cloud provider)

### CODE REVIEW

File: cluster/ack.go [hunk 1]
```diff
@@ -130,13 +132,24 @@ func (c *ACKCluster) SetTTL(ttl time.Duration) {
 func (c *ACKCluster) ListNodeNames() (pkgCommon.NodeNames, error) {
 	essClient, err := c.GetAlibabaESSClient(nil)
 	if err != nil {
-		return nil, err
+		return nil, emperror.Wrap(err, "failed to get Alibaba ESS client")
+	}
+
+	ecsClient, err := c.GetAlibabaECSClient(nil)
+	if err != nil {
+		return nil, emperror.Wrap(err, "failed to get Alibaba ECS client")
```
Comment (baluchicken) [inline cluster/ack.go]: This log message is duplicated, we already put this message to the log inside c.GetAlibabaECSClient

File: cluster/ack.go [hunk 2]
```diff
@@ -130,13 +132,24 @@ func (c *ACKCluster) SetTTL(ttl time.Duration) {
 func (c *ACKCluster) ListNodeNames() (pkgCommon.NodeNames, error) {
 	essClient, err := c.GetAlibabaESSClient(nil)
 	if err != nil {
-		return nil, err
+		return nil, emperror.Wrap(err, "failed to get Alibaba ESS client")
```
Comment (baluchicken) [inline cluster/ack.go]: This log message is duplicated, we already put this message to the log inside cGetAlibabaESSClient

File: helm/install.go [hunk 3]
```diff
@@ -355,3 +363,33 @@ func Install(helmInstall *phelm.Install, kubeConfig []byte) error {
 	log.Info("Helm install finished")
 	return nil
 }
+
+// watchTillerUntilReady waits for the tiller pod to become available.
+//
+// Returns true if it exists. If the timeout was reached and it could not find the pod, it returns false.
+func watchTillerUntilReady(namespace string, client *kubernetes.Clientset, timeout time.Duration, newImage string) bool {
+	deadlinePollingChan := time.NewTimer(timeout).C
```
Comment (baluchicken) [inline helm/install.go]: minor: we already have a library which handles backoff, https://github.com/banzaicloud/pipeline/tree/master/internal/backoff

Comment (stoader) [reply helm/install.go]: Removing this part as this is already handled in `WaitingForTillerComeUp`

File: pkg/providers/alibaba/provider.go [hunk 4]
```diff
@@ -14,4 +14,32 @@
...
 const Provider = "alibaba"
+
+// GetESSServiceEndpoint returns the endpoint of the ESS Service in the
+// given region (https://www.alibabacloud.com/help/doc-detail/25927.htm)
+func GetESSServiceEndpoint(region string) string {
+
+	region = strings.ToLower(region)
+
+	switch strings.ToLower(region) {
+	case "cn-zhangjiakou":
```
Comment (bonifaido) [inline pkg/providers/alibaba/provider.go]: Please add `fallthrough` statements.

**Pull request merged**